tests/test_inventory_builder.py

```diff
@@ -74,3 +99,174 @@ def test_expr_reso_grammar_ok(inv_builder, test_data, arg, exp):
 
     t = Transformer(data=test_data)
     assert t.transform(p.parse(arg)) == exp
+
+
+@pytest.mark.parametrize("args", [
+    ({  # config file plus list
+        "config-file": "netbox.yml",
+        "list": True,
+        "host": None,
+    }),
+    ({  # no list, plus host
+        "config-file": "netbox.yml",
+        "list": False,
+        "host": "hostname",
+    }),
+    ({  # list alone, no host, no config file
+        "config-file": "netbox.yml",
+        "list": True,
+        "host": None,
+    }),
+    ({  # list alone, no host, no config file
+        "config-file": "netbox.yml",
+        "list": False,
+        "host": None,
+    }),
+])
+def test_init_builder_ok(args, config):
+    class Args:
+        def __init__(self):
+            self.config_file = args['config-file']
+            self.host = args['host']
+            self.list = args['list']
+
+    builder = InventoryBuilder(Args(), config)
+
+    assert builder.config_file == args['config-file']
+    assert builder.list_mode == args['list']
+    assert builder.host == args['host']
+
+
+@pytest.mark.parametrize("hostvalue", [
+    None,
+    "",
+    "newshost",
+])
+def test_get_identifier_ok(cli_args, config, hostvalue):
+    import_type = "devices"
+    assert InventoryBuilder._get_identifier(
+        InventoryBuilder(cli_args, config),
+        dict(name=hostvalue),
+        import_type
+    )
+
+
+@pytest.mark.parametrize("element_name, group_name, inventory", [
+    (
+        "BK A07-0",
+        "dev_cpe",
+        {"_meta": {"hostvars": {}}}
+    ),
+    (
+        "amf-1.cpe0009",
+        "devices",
+        {"_meta": {"hostvars": {}}}
+    ),
+    (
+        "BK A07-0",
+        "all",
+        {"_meta": {"hostvars": {}}}
+    ),
+])
+def test_add_element_to_group_ok(cli_args, config, element_name, group_name, inventory):
+    assert InventoryBuilder._add_element_to_group(
+        InventoryBuilder(cli_args, config),
+        element_name,
+        group_name,
+        inventory
+    ) is None
+
+
+@pytest.mark.parametrize("group_name, inventory", [
```

Comment:
Same as above here, the structure of the inventory dict is not checked after _initialize_group call. I would suggest adding a parametrized variable, "expected", containing the final shape of the inventory dict.